includes/promoted-jobs/class-wp-job-manager-promoted-jobs-api.php

```diff
@@ -217,9 +217,19 @@ public function update_job_status( $request ) {
 	 */
 	public function get_job_data( $request ) {
 		$job_id = $request->get_param( 'job_id' );
-		if ( 'job_listing' !== get_post_type( $job_id ) ) {
+		$post   = get_post( $job_id );
+		if ( 'job_listing' !== get_post_type( $post ) ) {
 			return new WP_Error( 'not_found', __( 'The promoted job was not found', 'wp-job-manager' ), [ 'status' => 404 ] );
 		}
+		/**
+		 * The REST controller for job listings.
+		 *
+		 * @var WP_REST_Posts_Controller $controller
+		 */
+		$controller = get_post_type_object( 'job_listing' )->get_rest_controller();
+		if ( ! $controller->check_read_permission( $post ) ) {
```

Comment:
Officially `get_rest_controller()` returns a `WP_REST_Controller`, which doesn't have the `check_read_permission` method.

However, in `register_post_type` for the `job_listing` post type we explicitly define the `WP_REST_Posts_Controller` as the rest controller for this post type, so...i think we're safe here? Anyway I added the phpdoc just to avoid complaints from IDEs..

Reply:
I think we are pretty safe, but what do you think about checking the class type? Or maybe the existence of the `check_read_permission` method?

In case it isn't we can decide forbidding access (in order to not be less restrictive ever).

Reply:
After a second check, can't we use `get_item_permissions_check` ? Which is part of the `WP_REST_Controller` signature and seems like ends up checking `check_read_permission` in the WP_REST_Posts_Controller implementation?

Reply:
Yeah, sorry for the wording. It doesn't look very easy to understand now that I read it a second time haha

My thought was that, since this is to not ever leak more information than what REST API does. In the case we can't do the check, I would assume we want to fail.
But it was just a suggestion because you would have to make that decision. Feel free to go with the other option if you want. I don't think it will ever happen anyway.

Reply:
@aaronfc Oh, I understand you now, and yeah, it sounds fair. I adapted the check to block requests if the REST Controller isn't an instance of `WP_REST_Posts_Controller` in 5356390. Thanks!